core/src/main/java/org/opensearch/sql/expression/ReferenceExpression.java

```diff
@@ -100,7 +101,12 @@ public ExprValue resolve(ExprTupleValue value) {
   }
 
   private ExprValue resolve(ExprValue value, List<String> paths) {
-    final ExprValue wholePathValue = value.keyValue(String.join(PATH_SEP, paths));
+    ExprValue wholePathValue = value.keyValue(String.join(PATH_SEP, paths));
+    // For array types only first index currently supported.
```

Comment:
Please, submit a GH ticket to fix that once this PR merged.

Reply:
Could you elaborate a bit with an example?

Reply:
The current functionality for `OBJECT` arrays in the V2 engine is to return the first index of the array, while the legacy engine does not. See the changes I made to a returned `OBJECT` array in [partiql.rst](https://github.com/opensearch-project/sql/blob/df5ddb0a84ba259a3cf7d41d05a1e47f4b9d8b8e/docs/user/beyond/partiql.rst#example-1-selecting-top-level). Previously if a user did a query like `SELECT message.info FROM nested_objects;` where `message` was `ARRAY` type(NESTED), the query would fallback to the legacy engine for execution. Now that we support the `ARRAY` type with the nested implementation we need to support that type in the `SELECT` clause without the `NESTED` function. Returning the first item in the array aligns with the current V2 engine array support and without this change the returned result would always be `null` for `ARRAY` types.

Reply:
Already is a  GH issue for tracking arrays in V2 engine: [1300](https://github.com/opensearch-project/sql/issues/1300)